imap_processing/tests/ena_maps/test_ena_maps.py

```diff
@@ -836,3 +895,132 @@ def test_match_coords_to_indices_map_to_map_no_et_error(self):
+            assert skymap_from_dict.values_to_push_project == []
+            assert (
+                skymap_from_dict.values_to_pull_project
+                == skymap_props_dict["values_to_pull_project"]
+            )
+
+        for key in [
+            "sky_tiling_type",
+            "spice_reference_frame",
+            "nside",
+            "nested",
+            "spacing_deg",
+        ]:
+            if key in skymap_props_dict:
+                assert dict_from_skymap[key] == skymap_props_dict[key]
+
+        # Check that the dict from the SkyMap matches the original dict
+        assert dict_from_skymap == skymap_props_dict
+
+        # Change a value in the new dict and check that it is not equal to the original
+        dict_from_skymap["spice_reference_frame"] = "SPACE!"
+        assert (
+            dict_from_skymap["spice_reference_frame"]
+            != skymap_props_dict["spice_reference_frame"]
+        )
+
+    def test_to_json_and_from_json(self):
+        """Test serialization to and from JSON"""
+        # Make a SkyMap from the original properties dict
+        skymap_props_dict = {
+            "sky_tiling_type": "HEALPIX",
+            "nside": 32,
+            "nested": True,
+            "spice_reference_frame": geometry.SpiceFrame.ECLIPJ2000.name,
+            "values_to_push_project": ["foo", "bar"],
+        }
+
+        # Write a temporary json file with the properties dict
+
+        with tempfile.NamedTemporaryFile(
+            delete=False, suffix=".json", mode="w"
+        ) as temp_file:
+            json.dump(skymap_props_dict, temp_file)
+            temp_file_path_input = temp_file.name
+
+        # Read the json file and create a new SkyMap from it
+        skymap_from_json = ena_maps.AbstractSkyMap.from_json(temp_file_path_input)
+
+        # Create json output from the SkyMap at a separate temporary file path
+        temp_file_path_output = tempfile.NamedTemporaryFile(
+            delete=False, suffix=".json", mode="w"
+        ).name
+        skymap_from_json.to_json(json_path=temp_file_path_output)
+
+        assert skymap_from_json.spice_reference_frame == geometry.SpiceFrame.ECLIPJ2000
+        assert skymap_from_json.tiling_type is ena_maps.SkyTilingType.HEALPIX
+        assert skymap_from_json.nside == 32
+        assert skymap_from_json.nested is True
+        assert skymap_from_json.values_to_push_project == ["foo", "bar"]
+        assert skymap_from_json.values_to_pull_project == []
+
+        # Expect there to be a AttributeError when accessing a non-existent key
+        with pytest.raises(AttributeError):
+            _ = skymap_from_json.spacing_deg
+
+        # Check that the json output is the same as the original input
+        with open(temp_file_path_input) as f:
+            original_json = json.load(f)
+        with open(temp_file_path_output) as f:
+            output_json = json.load(f)
+        assert original_json == output_json
```

Comment:
You can combine this into a single context manager line if you'd like as well, but what you have is fine too.
```suggestion
        with open(temp_file_path_input) as f_in, open(temp_file_path_output) as f_out:
            assert json.load(f_in) == json.load(f_out)
```

Reply:
This is cleaner as you have it, so I'll use this in the future! Turns out I need to do a few _more_ things with the dicts loaded from json, so I think it's neater this way. 

```
# Check that the json output is the same as the original input ONLY
        # after adding automatically added
        # "values_to_push_project"/"values_to_pull_project" key to the original dict
        with open(temp_file_path_input) as f:
            original_json = json.load(f)
        with open(temp_file_path_output) as f:
            output_json = json.load(f)
        # The output json will have added an empty list for values_to_pull_project
        assert original_json != output_json
        # add the values_to_pull_project key to the original json
        original_json["values_to_pull_project"] = []
        assert original_json == output_json

```